tsdb/engine/tsm1/engine.go

```diff
@@ -475,6 +477,10 @@ func (e *Engine) WithLogger(log zap.Logger) {
 
 // LoadMetadataIndex loads the shard metadata into memory.
 func (e *Engine) LoadMetadataIndex(shardID uint64, index tsdb.Index) error {
+	if index.Type() != inmem.IndexName {
```

Comment:
Yep. I looked at making it seamless @jwilder with noop calls but it looked like a lot of trouble.